catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/XstreamPathConverter.java

```diff
@@ -0,0 +1,143 @@
+     * @throws ConversionException
+     */
+    @Override
+    public Object unmarshal(HierarchicalStreamReader reader, UnmarshallingContext context)
+            throws ConversionException {
+
+        if (pathValueTracker != null) {
+
+            PathTracker tracker = new PathTracker();
+            PathTrackingReader pathReader = new PathTrackingReader(reader, tracker);
+
+            readPath(pathReader, tracker);
+
+        }
+        return pathValueTracker;
+
+    }
+
+    /**
+     * Set the paths the convert should attempt to find
+     *
+     * @param paths
+     */
+    public void setPaths(List<Path> paths) {
+        pathValueTracker = new XstreamPathValueTracker();
+        pathValueTracker.buildPaths(paths);
+    }
+
+    /**
+     * Reads through the tree looking for a specific path and returns the value at that node
+     * <p/>
+     * The reader is moved to the next node in the path
+     * <p/>
+     * For example, if readPath(reader, "a", "b", "c") is called, then the value at /a/b/c is
+     * returned and the reader is advanced to "<d>"
+     * {code}
+     * <a>  <-- reader starts here
+     * <b>
+     * <c>value</c>
+     * </b>
+     * </a>
+     * <d></d>
+     * {code}
+     *
+     * @param reader
+     * @param tracker
+     */
+    private void readPath(PathTrackingReader reader, PathTracker tracker) {
+
+        updatePaths(reader, tracker.getPath());
+        while (reader.hasMoreChildren()) {
+            reader.moveDown();
+
+            readPath(reader, tracker);
+            reader.moveUp();
+        }
+    }
+
+    private void updatePaths(PathTrackingReader reader, Path currentPath) {
+        pathValueTracker.getPaths()
+                .forEach(path -> {
+
+                    if (doBasicPathsMatch(path, currentPath)) {
+
+                        if (path.toString()
+                                .contains("@")) {
+                            String attributeName = StringUtils.substringAfterLast(path.toString(),
+                                    "@");
+                            pathValueTracker.add(path, reader.getAttribute(attributeName));
+
+                        } else {
+                            pathValueTracker.add(path, reader.getValue());
+                        }
+
+                    }
+                });
+    }
+
+    private boolean doBasicPathsMatch(final Path path1, final Path path2) {
+        if (path1.equals(path2)) {
+            return true;
+        }
+        // ignore count designators and attribute specifications whenc omparing paths
+        // ie, /a/b[3]/c/@foo for our purposes is equivalent to /a/b/c
+
+        String path1Repalced = StringUtils.chomp(path1.toString()
+                .replaceAll("\\[.*\\]", "")
+                .replaceAll("/@.*$", ""), "/");
+        String path2Repalced = StringUtils.chomp(path2.toString()
+                .replaceAll("\\[.*\\]", "")
+                .replaceAll("/@.*$", ""), "/");
+
```

Comment:
Since regexes can be tricky, I'm going to throw out the idea of pulling these out and putting them into a utility class so unit tests can be written for each regex.